mono/metadata/threads.c

```diff
@@ -745,7 +745,14 @@ static guint32 WINAPI start_wrapper_internal(void *data)
 		args [0] = start_arg;
 		/* we may want to handle the exception here. See comment below on unhandled exceptions */
 		mono_runtime_delegate_invoke_checked (start_delegate, args, &error);
-		mono_error_raise_exception (&error); /* OK, triggers unhandled exn handler */
+
+		if (!mono_error_ok (&error)) {
+			MonoException *ex = mono_error_convert_to_exception (&error);
+			mono_error_cleanup (&error);
```

Comment:
You can also do:
ex = convert_to_exception ()
if (ex)
   mono_unhandled_exception (ex);